src/main/java/org/folio/rest/client/OkapiClient.java

```diff
@@ -44,4 +58,46 @@
       .putHeader(OKAPI_URL_HEADER, okapiUrl)
       .putHeader(OKAPI_HEADER_TOKEN, token);
   }
+
+  public <T> Future<T> getById(String resourcePath, String id, Class<T> objectType) {
+    Optional<String> validationError = validateGetByIdArguments(resourcePath, id, objectType);
+    if (validationError.isPresent()) {
+      String errorMessage = validationError.get();
+      log.error(errorMessage);
+      return failedFuture(new IllegalArgumentException(errorMessage));
+    }
+
+    Promise<HttpResponse<Buffer>> promise = Promise.promise();
+    okapiGetAbs(resourcePath + "/" + id).send(promise);
+
+    return promise.future().compose(response -> {
+      if (response.statusCode() != 200) {
+        return failedFuture(format("Failed to get %s by ID. Response status code: %s",
+          objectType.getSimpleName(), response.statusCode()));
+      }
+      try {
+        T object = objectMapper.readValue(response.bodyAsString(), objectType);
+        return succeededFuture(object);
+      } catch (IOException exception) {
+        return failedFuture(format("Failed to parse response for %s. Response body: %s",
+          objectType.getSimpleName(), response.bodyAsString()));
```

Comment:
please add URL logging for failed module

Reply:
done